borsar/utils.py

```diff
@@ -135,14 +135,28 @@ def _check_tmin_tmax(raw, tmin, tmax):
     return tmin, tmax, sfreq
 
 
-def valid_windows(raw, tmin=None, tmax=None, winlen=2., step=1.):
-    '''
-    Do uzupełnienia przez Nastię :)
-    '''
-    tmin, tmax, sfreq = _check_tmin_tmax(raw, tmin, tmax)
+def valid_windows(raw, tmin, tmax, winlen, step):
+
+    sfreq = raw.info['sfreq']
+    annot = raw.annotations
+    winlen = int(round(2 * sfreq))
+    step = int(step*sfreq)
+    s_tmin, s_tmax = int(round(tmin*sfreq)), int(round(tmax*sfreq))
+    n_windows = int((s_tmax - s_tmin - (winlen - step))/step)
+    valid = list()
+    from borsar.utils import detect_overlap
+    for win_idx in range(n_windows):
+        segment = [(win_idx + tmin)*step, (win_idx + tmin)*step + winlen]
+        if borsar.utils.detect_overlap(segment, annot, samples = True) > 0.0:
```

Comment:
1. importujesz już funkcję więc `detect_overlap`. Gdybyś importowała tylko borsar to tu byś pisała tak. A nie importujesz borsar.
2. rób `valid.append(detect_overlap(...) > 0.)` zamiast tego if-else

Reply:
Rób albo:
```python
from borsar.utils import detect_overlap
# ... wiele linijek niżej:
detect_overlap(segment, annot, samples=True)
```
albo:
```python
import borsar
# ... wiele linijek niżej:
borsar.utils.detect_overlap(segment, annot, samples=True)
```

Reply:
a, ok, po prostu widziałam to jak komentarz do czterech linijek :D